charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```diff
@@ -274,6 +274,8 @@ spec:
   ruleSelector:
     matchLabels:
       release: {{ $.Release.Name | quote }}
+{{ else if eq .Values.prometheus.prometheusSpec.ruleSelector nil }}
```

Comment:
```suggestion
{{ else if not (kindIs "invalid" .Values.prometheus.prometheusSpec.ruleSelector) }}
```

Lets use the same pattern as used on other values.

Reply:
This would bring it in line with the other fields, but note that this would be a breaking change. With the suggested change, it will now result in `null` when exclusively setting `prometheus.prometheusSpec.ruleSelectorNilUsesHelmValues=false`, where it currently results in `{}`

<img width="1098" height="250" alt="image" src="https://github.com/user-attachments/assets/2cb836d6-9921-4c5b-86c0-4d50df81b687" />

Is this acceptable?